src/index.js

```diff
@@ -19,664 +20,520 @@ const debug = (function () {
 })()
 
 class Masq {
-  constructor (appName, appDescription, appImageURL, options = {}) {
-    this.eventTarget = document.createElement('MasqLib')
-
-    this.appName = appName
-    this.appDescription = appDescription
-    this.appImageURL = appImageURL
-
-    // override config with constructor options
-    this.config = {
-      hubUrls: options.hubUrls ? options.hubUrls : jsonConfig.hubUrls,
-      masqAppBaseUrl: options.masqAppBaseUrl ? options.masqAppBaseUrl : jsonConfig.masqAppBaseUrl,
-      swarmConfig: options.swarmConfig ? options.swarmConfig : jsonConfig.swarmConfig
-    }
+  get eventTarget () {
+    return this.service.machine.context.eventTarget
+  }
+
+  constructor (appName, appDescription, appImageURL, config) {
+    const eventTarget = document.createElement('MasqLib')
+    this.machine = Machine({
```

Comment:
Wouldn't it be also better to also initialize the machine outside `this` class ? This would favor separation of concern between Masq, the exposed api, and the machine which is a skeleton across our lib